sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/jsonExpressions.scala

```diff
@@ -536,26 +536,31 @@ case class JsonToStructs(
       timeZoneId = None)
 
   override def checkInputDataTypes(): TypeCheckResult = schema match {
-    case _: StructType | ArrayType(_: StructType, _) =>
+    case _: StructType | ArrayType(_: StructType | _: AtomicType, _) =>
       super.checkInputDataTypes()
     case _ => TypeCheckResult.TypeCheckFailure(
-      s"Input schema ${schema.simpleString} must be a struct or an array of structs.")
+      s"Input schema ${schema.simpleString} must be a struct or " +
+        s"an array of structs or primitive types.")
   }
 
   @transient
-  lazy val rowSchema = schema match {
+  lazy val rowSchema: DataType = schema match {
```

Comment:
Not schema for row anymore. Maybe `dataSchema`?

Reply:
Why is it not a row schema? It is, but sometimes the schema of a row is an array.

Reply:
I think it was row scheme because it can only be `StructType` before. This is not the input/output row's schema.

Reply:
I can't understand the difference. Anyway, if you think `dataSchema` is more appropriate, I can rename it.